x-pack/plugins/observability/public/typings/fetch_overview_data/index.ts

```diff
@@ -24,17 +25,29 @@ export interface FetchDataParams {
   absoluteTime: { start: number; end: number };
   relativeTime: { start: string; end: string };
   bucketSize: string;
+  serviceName?: string;
 }
 
+export interface HasDataParams {
+  absoluteTime: { start: number; end: number };
+}
+
+interface UXHasDataResponse {
+  hasData: boolean;
+  serviceName: string | number | undefined;
```

Comment:
I don't know about this, I feel `hasData` has been given way more responsibility than only checking if there is any data available.

Reply:
with separation of hooks this will improve